packages/lisk-transactions/src/base_transaction.ts

```diff
@@ -0,0 +1,263 @@
+	}
+
+	public abstract prepareTransaction(
+		passphrase: string,
+		secondPassphrase?: string,
+	): TransactionJSON;
+
+	public toJSON(): TransactionJSON {
+		const transaction = {
+			id: this.id,
+			amount: this.amount.toString(),
+			type: this.type,
+			timestamp: this.timestamp,
+			senderPublicKey: this.senderPublicKey,
+			senderId: this.senderId,
+			recipientId: this.recipientId,
+			recipientPublicKey: this.recipientPublicKey,
+			fee: this.fee.toString(),
+			signatures: this.signatures,
+			asset: this.asset,
+		};
+
+		if (!(typeof this.signature === 'string' && this.signature.length > 0)) {
+			return transaction;
+		}
+
+		const singleSignedTransaction = {
+			...transaction,
+			signature: this.signature,
+		};
+
+		if (
+			!(typeof this.signSignature === 'string' && this.signSignature.length > 0)
+		) {
+			return singleSignedTransaction;
+		}
+
+		const signedTransaction = {
+			...singleSignedTransaction,
+			signSignature: this.signSignature,
+		};
+
+		return signedTransaction;
+	}
+
+	public getBytes(): Buffer {
+		const { signature, signSignature } = this.toJSON();
+		const transactionType = Buffer.alloc(BYTESIZES.TYPE, this.type);
+		const transactionTimestamp = Buffer.alloc(BYTESIZES.TIMESTAMP);
+		transactionTimestamp.writeIntLE(this.timestamp, 0, BYTESIZES.TIMESTAMP);
+
+		const transactionSenderPublicKey = cryptography.hexToBuffer(
+			this.senderPublicKey,
+		);
+
+		const transactionRecipientID = this.recipientId
+			? cryptography.bigNumberToBuffer(
+					this.recipientId.slice(0, -1),
+					BYTESIZES.RECIPIENT_ID,
+			  )
+			: Buffer.alloc(BYTESIZES.RECIPIENT_ID);
+
+		const amountBigNum = new BigNum(this.amount);
+
+		const transactionAmount = amountBigNum.toBuffer({
+			endian: 'little',
+			size: BYTESIZES.AMOUNT,
+		});
+
+		const transactionSignature = signature
+			? cryptography.hexToBuffer(signature)
+			: Buffer.alloc(0);
+
+		const transactionSecondSignature = signSignature
+			? cryptography.hexToBuffer(signSignature)
+			: Buffer.alloc(0);
+
+		return Buffer.concat([
+			transactionType,
+			transactionTimestamp,
+			transactionSenderPublicKey,
+			transactionRecipientID,
+			transactionAmount,
+			transactionSignature,
+			transactionSecondSignature,
+		]);
+	}
+
+	public abstract containsUniqueData(): boolean;
+
+	public validate(): ValidateReturn {
+		const transaction = this.toJSON();
+
+		if (!TRANSACTION_TYPES.includes(transaction.type)) {
+			return {
+				validated: false,
+				errors: [
+					new TransactionError('Invalid transaction type.', transaction.id),
+				],
+			};
+		}
+
+		// Schema validation
+		const { valid, errors } = validateTransaction(transaction);
+		const transactionErrors = errors
+			? errors.map(
+					error =>
+						new TransactionError(
+							`'${error.dataPath}' ${error.message}`,
+							transaction.id,
+							error.dataPath,
+						),
+			  )
+			: undefined;
+
+		// Single signature validation
+		if (!transaction.signature) {
+			return {
```

Comment:
why are we returning single error here?

Reply:
@shuse2 updated to not use `utils/verifyTransaction` and to return all errors at the end